docs/contributing/documentation/overview.rst

```diff
@@ -33,14 +41,43 @@ Create a dedicated branch for your changes (for organization):
 
     $ git checkout -b docs/improving_foo_and_bar
 
-You can now make your changes directly to this branch and commit them. When
-you're done, push this branch to *your* GitHub fork and initiate a pull request.
+You can now make your changes directly to this branch and commit them.
+Remember to name you commits descriptively, keep them possibly small, with just unitary changes (such that change something only in one part of the docs, not everywhere).
+
+When you're done, push this branch to *your* GitHub fork and initiate a pull request.
+
+Your pull request will be reviewed, you will be asked to apply fixes if necessary and then it will be merged into the main repository.
+
+
+Testing Documentation
```

Comment:
IMO this section should reflect https://github.com/Sylius/Sylius/tree/master/docs#build.